### DISCUSSION TITLE
ZipStream v3 Beta

### DISCUSSION THREAD

**Discussion by maennchen** [spatie/laravel-medialibrary#3129] [General] [answered]
Hi All :wave: 

Since this package is one of the main dependents on `maennchen/zipstream-php` and we just released a beta of v3, I would like to ask you for some feedback before we deploy the version as stable.

Since v3 will only support PHP 8.1 and up, you will probably want to wait until PHP 8.0 is no longer supported to use it.

* Discussion for feedback: https://github.com/maennchen/ZipStream-PHP/discussions/237
* Upgrade instructions: https://github.com/maennchen/ZipStream-PHP/tree/3.0.0-beta.2#upgrade-to-version-300

This release mainly focuses on internal code quality, which should make it easier for outside people to contribute to the project. The code (and the underlying standard) was quite cryptic until now which prevented most people from doing so.

Thanks!

**Comment by freekmurze** [comment, marked as answer]
I'll try to test with your beta somewhere next week. Thanks for all the work you put in that package 👍

**Reply by freekmurze** [reply]
I can confirm that our test suite reports no errors when using your latest beta 👍 

Just had to do a small refactor due to the remove `ZipStream\Option\Archive`.

**Reply by freekmurze** [reply]
This refactor is a breaking change for us, so we'll tag a new major release when we switch to v3 op ZipStream. Probably we'll tag a new release in tandem with Laravel 10 early February.
